### PR TITLE
refactor: Use a Credentials struct consistently

### PR DESCRIPTION
**Pull request by dlipovetsky**
**What problem does this PR solve?**:
Previously, we used a Credentials struct in some places, and a LocalObjectReference in others.  With this change, we use a Credentials struct in all places.

**Which issue(s) this PR fixes**:
Fixes #

**How Has This Been Tested?**:

**Special notes for your reviewer**:

### CODE REVIEW

File: api/v1alpha1/addon_types.go [hunk 1]
```diff
@@ -174,11 +174,23 @@ type StorageClassConfig struct {
 	AllowExpansion bool `json:"allowExpansion,omitempty"`
 }
 
+type CSICredentials struct {
```
Comment (dkoshkin) [inline api/v1alpha1/addon_types.go:177]: I think the examples need to change too https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/tree/main/examples/capi-quick-start, and possible some snippets in docs

**Review by dkoshkin — COMMENTED** [review]
Thanks for making it more explicit

File: api/v1alpha1/addon_types.go (hunk 1 above)
Comment (dlipovetsky) [reply api/v1alpha1/addon_types.go:177]: Good catch, I'll update that

Comment (dlipovetsky) [reply api/v1alpha1/addon_types.go:177]: Fixed in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/663/commits/18dbf4def55e390325a1f056a3abd9c40aeaea50

**Review by dkoshkin — APPROVED** [review]

**Review by supershal — APPROVED** [review]
Thank you keep fixing existing consistency issues.

**Pull request merged**